crawlerr/services/strategies/base_strategy.py

```
@@ -0,0 +1,72 @@
+
+        Args:
+            request: Strategy-specific request object
+
+        Returns:
+            True if request is valid
+        """
+        pass
+
+    async def pre_execute_setup(self) -> None:
+        """Setup tasks to run before crawling begins."""
+        pass
```

Ruff (0.12.2)

`BaseCrawlStrategy.pre_execute_setup` is an empty method in an abstract base class, but has no abstract decorator

(B027)